app/controllers/two_factor_authentication/otp_expired_controller.rb

```diff
@@ -0,0 +1,13 @@
+module TwoFactorAuthentication
+  class OtpExpiredController < ApplicationController
```

Comment:
We may want to have some request filtering to ensure that the user is authenticated, since we're referencing `current_user`. Otherwise, this route could produce 500 errors if someone were to navigate to it directly.

e.g.

```rb
  before_action :confirm_two_factor_authenticated
```

Reply:
The validation that was added will handle the case that the user is logged in but shouldn't be seeing this screen, but I can still navigate to this page while logged out and hit a 500 error:

![image](https://user-images.githubusercontent.com/1779930/214595246-be32f1d5-8174-4924-b294-36113a912b3f.png)

Reply:
I see the challenge here since the default `:confirm_two_factor_authenticated` helper I recommended in my first comment would also prevent the user from visiting this page during the MFA authentication step.

I think we could use a conditional to only enforce it unless there's a (not-necessarily-fully-authenticated) user?

```suggestion
module TwoFactorAuthentication
  class OtpExpiredController < ApplicationController
    before_action :confirm_two_factor_authenticated, unless: :user_signed_in?
```